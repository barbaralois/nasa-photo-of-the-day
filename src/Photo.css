.photo-list img {
  height: 120px;
  border-radius: 5px;
  border: 1px solid white;
}
