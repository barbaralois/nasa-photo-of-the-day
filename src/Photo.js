import React from 'react';
import './Photo.css';

export default function Photo(props) {
  const { setAPOD } = props;

  const clickHandler = (event) => {
    setAPOD(props.index);
  };

  return <img src={props.imgSrc} alt={props.date} onClick={clickHandler} />;
}
