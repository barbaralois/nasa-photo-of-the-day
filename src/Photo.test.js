import React from 'react';
import Photo from './Photo';
import renderer from 'react-test-renderer';

it('renders correctly when there are no items', () => {
  const tree = renderer
    .create(
      <Photo
      // imgSrc={photo.hdurl}
      // date={photo.date}
      // key={index}
      // index={index}
      // setAPOD={props.setAPOD}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with props', () => {
  const testURL = 'www.google.com';
  const testDate = '5-15';
  const testKey = 5;
  const testIndex = 5;
  const setter = 'a function';
  const tree = renderer
    .create(
      <Photo
        imgSrc={testURL}
        date={testDate}
        key={testKey}
        index={testIndex}
        setAPOD={setter}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
