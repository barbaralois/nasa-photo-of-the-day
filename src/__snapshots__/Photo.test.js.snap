// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<img
  className="sc-AxirZ fXuOHl"
  onClick={[Function]}
/>
`;

exports[`renders correctly with props 1`] = `
<img
  alt="5-15"
  className="sc-AxirZ fXuOHl"
  onClick={[Function]}
  src="www.google.com"
/>
`;
